policies/design-process.md

```diff
@@ -0,0 +1,141 @@
+primary characteristics of the solution approach so that it can be compared
+against the other candidate solutions and to determine how well it meets the
+requirements and constraints.
+
+In some cases a candidate solution may be identified that meets most of the
+requirements and constraints but not all. Where this happens the requirements
+and constraints that are not met by the candidate solution should be clearly
+stated.
+
+Typically the design team will be the people that identify and document the
+candidate solutions. In some cases it may be appropriate to hold a design
+workshop to collectively identify the candidate solutions. In this case it will
+typically still be necessary for the design team to document the solution
+options outside of the workshop.
+
+The design team may choose to make recommendations about the preferred solution
+option.
+
+### Select solution
+
+Once the candidate solutions have been identified and documented a solution
+must be selected by the OTC. There should be an opportunity for the OpenSSL
+community or non-OTC members to provide input into the decision. This could
+either be in the form of comments on a draft Pull Request describing the
+candidate solutions, or it could be in the form of an open invite for community
+members to join the workshop where the solution is selected.
+
+Optionally the OTC could select no solution and instead direct the design team
+to iterate over the candidate solutions again and either make refinements or
+identify additional options.
+
+### Refine and document
+
+In this step the design team provide further detail on the selected solution
+option. There should be sufficient detail to enable the implementation work to
+progress and to understand the trajectory of the solution into the future. Some
+details may be omitted until a later stage of the development project if
+appropriate and they are not needed until later.
+
+The design document should be available for public comment in the form of a
+pull request.
+
+### Agree solution
+
+In this step the OTC agree the solution as it is described in the design
+document as produced by the design team. Once agreed the implementation may
+proceed and the design document should be merged into the otc repository.
+
+The OTC may choose not to agree a solution. In this case the process should
+iterate back to the "Refine and document" step to incorporate feedback from the
+OTC. In some cases more significant rework may be required, in which case the
+OTC may choose to step back to any previous step in this design process.
+
+In some cases the OTC may agree a solution which has certain details
+deliberately omitted in order to defer work on those details until a later stage
+of the implementation project. In such cases this process iterates back to the
+"Refine and document" stage to fill in those details when they are eventually
+required, and the subsequent amended design document must be agreed again by the
+OTC.
+
```

Comment:
We also need some sort of a review solution. The original design could miss some crucial points or see a better approach, so the design will require some changes.

Reply:
I saw this as happening on github in the pull request, i.e. in the "refine and document" step the design document is made available as a PR on which people can make comments. The document is updated accordingly to take account of comments. Eventually the document stabilises sufficiently that OTC are asked to agree it. Either they agree, or they think it needs more work in which case it goes back to "refine and document" as stated above, i.e. "The OTC may choose not to agree a solution. In this case the process should iterate back to the "Refine and document" step to incorporate feedback from the OTC."